transactions/src/main/java/org/ehcache/transactions/xa/XAStore.java

```diff
@@ -838,7 +842,12 @@ public void releaseStore(Store<?, ?> resource) {
         // release the underlying store first, as it may still need the serializer to flush down to lower tiers
         underlyingStoreProvider.releaseStore(xaStore.underlyingStore);
         try {
-          helper.softLockSerializerRef.getAndSet(null).close();
+          Serializer<?> serializer = helper.softLockSerializerRef.getAndSet(null);
```

Comment:
There is some weird code around type here - the `Serializer` held by the `AtomicReference` is always of a  single type that does not need to be closed.
I recommend changing the reference to be of the more specific type `SoftLockSerializer` and thus dropping the if block after it because that other guy does not need to be `Closeable`.